src/scripts/main.js

```diff
@@ -1,3 +1,48 @@
 'use strict';
 
-// write your code here
+let bodyDoc;
+let mateLogo;
+
+if (document.querySelector('body') !== null) {
+  bodyDoc = document.querySelector('body');
+}
+
+if (document.querySelector('.logo') !== null) {
+  mateLogo = document.querySelector('.logo');
+}
+
+const promise1 = () => {
+  return new Promise((resolve, reject) => {
+    mateLogo.addEventListener('click', () => resolve('Promise was resolved!'), {
+      once: true,
+    });
+  });
+};
+
+const promise2 = () => {
+  return new Promise((resolve, reject) => {
+    // eslint-disable-next-line prefer-promise-reject-errors
+    setTimeout(() => reject('Promise was rejected!'), 3000);
+  });
+};
+
+function makeDiv(className) {
+  const div = document.createElement('div');
+
+  div.setAttribute('class', className);
+  bodyDoc.appendChild(div);
+
+  return div;
+}
+
+promise1().then((val) => {
+  const successHandler = makeDiv('message');
+
+  successHandler.textContent = val;
+});
+
+promise2().catch((val) => {
+  const errorHandler = makeDiv('message error-message');
+
+  errorHandler.textContent = val;
+});
```

Comment:
There is no success handling for the promise returned by 'promise2'. If it's intended to only handle errors, this might be fine, but typically you would also want to handle the successful resolution of a promise.